jmh-benchmarks/src/main/java/org/apache/kafka/jmh/assignor/TargetAssignmentBuilderBenchmark.java

```diff
@@ -0,0 +1,202 @@
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Threads;
+import org.openjdk.jmh.annotations.Warmup;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+@State(Scope.Benchmark)
+@Fork(value = 1)
+@Warmup(iterations = 5)
+@Measurement(iterations = 5)
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+public class TargetAssignmentBuilderBenchmark {
+
+    @Param({"100", "500", "1000", "5000", "10000"})
+    private int memberCount;
+
+    @Param({"5", "10", "50"})
+    private int partitionsToMemberRatio;
+
+    @Param({"10", "100", "1000"})
+    private int topicCount;
+
+    private static final String GROUP_ID = "benchmark-group";
+
+    private static final int GROUP_EPOCH = 0;
+
+    private PartitionAssignor partitionAssignor;
+
+    private Map<String, TopicMetadata> subscriptionMetadata = Collections.emptyMap();
+
+    private TargetAssignmentBuilder targetAssignmentBuilder;
+
+    private AssignmentSpec assignmentSpec;
+
+    private final List<String> allTopicNames = new ArrayList<>(topicCount);
+
+    private final List<Uuid> allTopicIds = new ArrayList<>(topicCount);
+
+    @Setup(Level.Trial)
+    public void setup() {
+        // For this benchmark we will use the Uniform Assignor
+        // and a group that has a homogeneous subscription model.
+        partitionAssignor = new UniformAssignor();
+
+        subscriptionMetadata = generateMockSubscriptionMetadata();
+        Map<String, ConsumerGroupMember> members = generateMockMembers();
+        Map<String, Assignment> existingTargetAssignment = generateMockInitialTargetAssignment();
+
+        // Add a new member to trigger a rebalance.
+        Set<String> subscribedTopics = new HashSet<>(subscriptionMetadata.keySet());
+
+        ConsumerGroupMember newMember = new ConsumerGroupMember.Builder("new-member")
+            .setSubscribedTopicNames(new ArrayList<>(subscribedTopics))
+            .build();
+
+        targetAssignmentBuilder = new TargetAssignmentBuilder(GROUP_ID, GROUP_EPOCH, partitionAssignor)
+            .withMembers(members)
+            .withSubscriptionMetadata(subscriptionMetadata)
+            .withTargetAssignment(existingTargetAssignment)
+            .addOrUpdateMember(newMember.memberId(), newMember);
+    }
+
+    private Map<String, ConsumerGroupMember> generateMockMembers() {
+        Map<String, ConsumerGroupMember> members = new HashMap<>();
+
+        for (int i = 0; i < memberCount - 1; i++) {
+            Set<String> subscribedTopics;
+            subscribedTopics = new HashSet<>(allTopicNames);
```

Comment:
nit: You can combine these two into one line.

Reply:
ah yes, I missed changing that